app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt

```diff
@@ -131,7 +131,7 @@ class BookmarkFragment : Fragment(), BackHandler, AccountObserver {
 
                 activity?.run {
                     ViewModelProviders.of(this).get(BookmarksSharedViewModel::class.java)
-                }!!.selectedFolder = null
+                }!!.selectedFolder = currentRoot
```

Comment:
@colintheshots why were we nulling this out in the first place? Doesn't it make sense to keep the ViewModel up to date with the root view so that when I press new folder, it knows which parent root to put it under?

Reply:
So this shared ViewModel is being reset each time we resume the bookmarkFragment. The user has not selected anything yet. The thinking was that selecting a destination folder for an existing bookmark item should be an affirmative choice by the user. Otherwise, you'd go to the folder selection screen from editing a bookmark and have a stale selection already in place.

The main purpose of the selectedFolder was to remember on returning in the back stack to the Edit Bookmark screen (from the Bookmark Folder Selection screen) where to move the current bookmark to.

It seems that your code here doesn't technically need to remember a selected folder between multiple fragments. One could pass a fragment argument, which wasn't an option when going backwards through the fragment back stack to the Edit Bookmark screen. But this could work, too. It is a UX change, but seems okay.